sdk/go/common/util/gitutil/git.go

```diff
@@ -255,9 +302,79 @@ func parseAuthURL(remoteURL string) (string, *http.BasicAuth, error) {
 	return u.String(), auth, nil
 }
 
+// sshConfig allows tests to mock SSH key paths.
+type sshConfig struct {
+	path string
+	err  error
+}
+
+// GetKeyPath reads SSH configuration from disk when the receiver is nil. When
+// non-nil it acts as a mock.
+func (c *sshConfig) GetKeyPath(host string) (string, error) {
+	if c == nil {
+		return ssh_config.GetStrict(host, "IdentityFile")
+	}
+	return c.path, c.err
+}
+
+// getSSHPublicKeys reads from the user's SSH configuration and returns public
+// keys for the given host.
+//
+// The `PULUMI_GITSSH_PASSPHRASE` environment variable can be provided if the
+// relevant key is passphrase protected, or (if in an interactive session) the
+// user will be prompted to input a passphrase.
+//
+// TODO: Integrate with GCM when https://github.com/go-git/go-git/issues/490
+// lands.
+//
+// This method handles `~/.ssh/config`, `/etc/host/ssh`, and `Include`
+// directives in the SSH configuration as you would expect.
+func getSSHPublicKeys(user string, host string, sshConfig *sshConfig) (*git_ssh.PublicKeys, error) {
+	privateKeyPath, err := sshConfig.GetKeyPath(host)
+	if err != nil {
+		return nil, err
+	}
+	// Expand tilde (~) if present in the path.
+	privateKeyPath, err = homedir.Expand(privateKeyPath)
```

Comment:
Weird, my workspace must not be set up correctly because my editor thought we were already using it as a dependency. Took it out and replaced it with this method -- thanks!